core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -64,86 +74,84 @@ oppia.controller('CollectionEditor', ['$scope', 'CollectionBackendApiService',
-
   // Stores a pending list of changes.
   $scope.changeList = [];
 
-  var _arrayContainsCaseInsensitive = function(array, element) {
-    var lowerElement = element.toLowerCase();
-    for (var i = 0; i < array.length; i++) {
-      if (array[i].toLowerCase() === lowerElement) {
+  $scope.addChange = function(change) {
+    $scope.changeList.push(change);
+  };
+
+  // TODO(mgowano): Handle a case where adding a pre-requisite skill
+  // will result into having all exploration requiring a pre-requisite
+  // skill. At the moment this is only handled when a change is committed
+  // to the backend.
+  $scope.addPrerequisiteSkill = function(skillName, expId) {
+    if (skillName && expId) {
+      var collectionNode = _getCollectionNodeForExplorationId(expId);
+      var prerequisiteSkills = collectionNode.prerequisite_skills;
+      if (!_arrayContainsCaseInsensitive(prerequisiteSkills, skillName)) {
+        prerequisiteSkills.push(skillName);
         return true;
       }
     }
     return false;
   };
 
-  $scope.addExploration = function(newExpId) {
-    if (newExpId) {
-      var change = CollectionUpdateService.buildAddCollectionNodeUpdate(
-        newExpId);
-      $scope.changeList.push(change);
-    }
-  };
-
-  $scope.deleteExploration = function(removeExpId) {
-    if (removeExpId) {
-      var change = CollectionUpdateService.buildDeleteCollectionNodeUpdate(
-        removeExpId);
-      $scope.changeList.push(change);
+  $scope.addAcquiredSkill = function(skillName, expId) {
+    if (expId && skillName) {
+      var collectionNode = _getCollectionNodeForExplorationId(expId);
+      var acquiredSkills = collectionNode.acquired_skills;
+      if (!_arrayContainsCaseInsensitive(acquiredSkills, skillName)) {
+        acquiredSkills.push(skillName);
+        return true;
+      }
     }
+    return false;
   };
 
-  $scope.setCollectionTitle = function(newCollectionTitle) {
-    if (newCollectionTitle) {
-      var change = CollectionUpdateService.buildCollectionTitleUpdate(
-        newCollectionTitle);
-      $scope.changeList.push(change);
+  // TODO(mgowano): Handle a case where deleting an exploration will
+  // result into unreachable exploration. At the moment this is only
+  // handled when a change is committed to the backend.
+  $scope.delExploration = function(expId) {
+    if (expId) {
+      for (var i = 0; i < $scope.collection.nodes.length; i++) {
+        var collectionNode = $scope.collection.nodes[i];
+        if (collectionNode.exploration_id == expId) {
+          $scope.collection.nodes.splice(i, 1);
+          return true;
+        }
+      }
     }
+    return false;
   };
 
-  $scope.setCollectionCategory = function(newCollectionCategory) {
-    if (newCollectionCategory) {
-      var change = CollectionUpdateService.buildCollectionCategoryUpdate(
-        newCollectionCategory);
-      $scope.changeList.push(change);
-    }
+  //TODO(mgowano): Implement adding a new exploration
```

Comment:
Please resolve this TODO. Maybe looking at how the backend handles this will help: